### ISSUE TITLE
Anyway to clear log and disable log for sometime ?

### ISSUE THREAD

**Issue by shirshak55** [laravel/telescope#20]
There is `telescope:prune` which i though will prune the logs but it is not working .

And there is someway to disable logging untill we specify start logging or like that. There is button that will prevent from loading but it still catches logs.

Thanks Themsaid again :)

**Comment by themsaid** [comment]
This command removes entries that are older than 24 hours

What do you mean by disable logging?

**Comment by vesper8** [comment]
It would be nice to be able to prune from the interface, including being able to prune ALL and not just the last 24 hours

It would also be nice to be able to disable telescope completely from the interface. I noticed there isn't a configuration variable in telescope.php to disable telescope

I kind of feel like telescope:install should publish the configuration file too

**Comment by shane-smith** [comment]
Try updating `config/telescope.php` and setting 'limit' to 0 (zero) before running the prune command? Untested.

**Comment by shirshak55** [comment]
@themsaid i mean i only want to start seeing events, queries when I explicitly on by clicking button so i can check it from fresh :)

Yes by pruning i mean able to prune all from interface not just last 24 hour.

**Comment by vesper8** [comment]
> If you carefully choose a different `database` connection for Telescope, you can easily 'prune' it by running `php artisan migrate:fresh --database=telescope`

or simply create a console command that truncates the telescope tables.. while this is an easy 'user fix'..  it would still be nice if this was a built-in feature

**Comment by shirshak55** [comment]
@warrenca  it would be better if ui provide it isn't it? It would be nice to have built in feature.

**Comment by warrenca** [comment]
@shirshak55 Yes I agree with that, but also consider that this is the first release and it's even on beta, I suppose. More features are coming out for sure.

For now I presented a work around to the feature you, me and others want. And besides, it's open source 😉, you know what I mean.

**Comment by 120dev** [comment]
Easy method by creating a command

`php artisan make:command telescope:clear`

in **`app/Console/Commands/telescope_clear.php`**

```
<?php

namespace App\Console\Commands;

use Illuminate\Console\Command;
use Illuminate\Support\Facades\DB;

class telescope_clear extends Command
{
    /**
     * The name and signature of the console command.
     *
     * @var string
     */
    protected $signature = 'telescope:clear';

    /**
     * The console command description.
     *
     * @var string
     */
    protected $description = 'Clear Telescope log';

    /**
     * Create a new command instance.
     *
     * @return void
     */
    public function __construct()
    {
        parent::__construct();
    }

    /**
     * Execute the console command.
     *
     * @return mixed
     */
    public function handle()
    {
        DB::statement('TRUNCATE telescope_entries CASCADE');
        DB::statement('TRUNCATE telescope_entries_tags CASCADE');
        DB::statement('TRUNCATE telescope_monitoring CASCADE');
    }
}
```

In **`app/Console/Kernel.php`**
add `telescope_clear::class;` in `commands()`

For more info to create a command : https://laravel.com/docs/5.7/artisan#writing-commands

**In terminal :**

` php artisan telescope:clear`

**Comment by MichMich** [comment]
I agree with others. When I tried telescope, first thing I looked for was a "clear" button in the UI. For example:

- Press clear button.
- Reload page.
- See which queries are executed.

Showing entries from previous sessions doesn't make sense most of the time. 

Adding a "pause logging" feature to the UI would be very welcome too!

**Comment by vesper8** [comment]
@120dev'd command didn't work for me but this did:
```
        Schema::disableForeignKeyConstraints();
        DB::connection('www')->statement('TRUNCATE telescope_entries');
        DB::connection('www')->statement('TRUNCATE telescope_entries_tags');
        DB::connection('www')->statement('TRUNCATE telescope_monitoring');
```

**Comment by shirshak55** [comment]
@themsaid  👍 here is your thank you 👍 

Thanks again :)

**Comment by fylzero** [comment]
You can use...

```
php artisan telescope:prune --hours=0
```

By default `telescope prune` will clear out everything from 48 hours back.  Use the command above to clear everything completely.
